storage/s3/s3.go

```diff
@@ -34,18 +36,55 @@ type minioStorage struct {
 }
 
 // New returns a new Storage that can store objects to S3.
-func New(bucket string, prefix string, metafilesPrefix string, mc MinioClient, l log.Logger) storage.Storage {
-	return &minioStorage{
-		bucket:          bucket,
-		mc:              mc,
-		l:               l,
-		prefix:          prefix,
-		metafilesPrefix: metafilesPrefix,
-		useDone:         metafilesPrefix != "",
+func New() storage.Storage {
+	return &minioStorage{}
+}
+
+const defaultEndpoint = "s3.amazonaws.com"
+
+type sourceConfig struct {
+	Endpoint        string
+	Insecure        bool
+	AccessKeyID     string `json:"accessKeyID"`
+	SecretAccessKey string
+	Bucket          string
+	Prefix          string
+	Recursive       bool
+}
+
+type destinationConfig struct {
+	sourceConfig    `mapstructure:",squash"`
+	MetafilesPrefix string
+}
+
+func (ms *minioStorage) Configure(config map[string]interface{}) error {
```

Comment:
I don't think this should live here. This package is _only_ about implementing the `storage.Storage` interface. IMO, this means that the destination plugin should be implemented in `plugins/s3` along with the S3 source plugin

Reply:
same topic https://github.com/connylabs/ingest/pull/47#discussion_r1024943162